src/engine/Server.cpp

```diff
@@ -256,13 +253,10 @@ ad_utility::url_parser::ParsedRequest Server::parseHttpRequest(
             R"(Request must only contain one of "query" and "update".)");
       }
       extractQueryFromParameters();
-      if (parsedRequest.parameters_.contains("update")) {
-        if (parsedRequest.parameters_["update"].size() > 1) {
-          throw std::runtime_error(
-              "The parameter \"update\" must be set exactly once.");
-        }
-        parsedRequest.operation_ =
-            Update{parsedRequest.parameters_["update"].front()};
+      auto update = ad_utility::url_parser::getParameterCheckAtMostOnce(
+          parsedRequest.parameters_, "update");
+      if (update.has_value()) {
+        parsedRequest.operation_ = Update{update.value()};
         parsedRequest.parameters_.erase("update");
```

Comment:
Now you can still implement

`auto setOperationIfSpecifiedInParams = [&]<typename Operation>(string_view paramName) {...}`

s.t. you here can write `setOperationIf....template operator()<Update>("update")`.
And reuse the code for `get` above.